neqo-transport/src/path.rs

```diff
@@ -98,48 +98,6 @@ impl Paths {
             })
     }
 
-    /// Find the path, but allow for rebinding.  That matches the pair of addresses
-    /// to paths that match the remote address only based on IP addres, not port.
-    /// We use this when the other side migrates to skip address validation and
-    /// creating a new path.
```

Comment:
Why did we previously skip address validation on port change?

If I understand RFC 9000 correctly, the port is always part of the validation process:

> In path validation, endpoints test reachability between a specific local address and a specific peer address, where an address is the 2-tuple of IP address and port.

https://www.rfc-editor.org/rfc/rfc9000.html#section-8.2

Reply:
@martinthomson might know?

Reply:
We did that deliberately, because if a peer changes just the port, it's probably just a NAT rebinding.  No point in wasting the effort.

Reply:
Do we care that that will make us violate the spec and fail a QNS test?